src/main/kotlin/org/rust/ide/folding/RustFoldingBuilder.kt

```diff
@@ -72,6 +72,15 @@ class RustFoldingBuilder() : FoldingBuilderEx(), DumbAware {
                 }
             }
 
+            override fun visitFnItem(o: RustFnItemElement) {
```

Comment:
Yes, the behaviour changed after adding `RustPathNameElement` to `RustFnItem`. So, it's covered by several (four to be exact) existing tests.